test/e2e/storage/pd.go

```diff
@@ -512,6 +536,37 @@ func detachPD(nodeName types.NodeName, pdName string) error {
 	}
 }
 
+func attachPD(nodeName types.NodeName, pdName string) error {
```

Comment:
agreed. I added a TODO here. The thing is - if we do move these to providers then somebody will have to implement these for other cloudproviders too, but it can be done incrementally I think.

Reply:
The only implemented provider apart from AWS and GCE is Azure, other providers are not implemented at all or use NullProvider, so it's not that complicated